### DISCUSSION TITLE
Seeking maintenance help on kdl4j and kq

### DISCUSSION THREAD

**Discussion by hkolbeck** [kdl-org/kdl#102] [General]
Hi folks,

I've been watching in some distress as changes to the language spec roll through, as I currently find myself unable to find the time to update [kdl4j](https://github.com/hkolbeck/kdl4j) and the jq-lite-like command line utility I built on top of it, [kq](https://github.com/hkolbeck/kq).

If anyone is interested in contributing to either, it would be immensely appreciated.

Thanks,
Hannah

**Comment by aulisius** [comment]
came here from twitter. @hkolbeck I'd like to help out with the kdl4j project. kq looks interesting but sadly I don't have much Kotlin experience.

**Comment by sthagen** [comment]
I am interested in contributing to both and I will take a look, Hannah.
First I will try to assess what is there, what already has been suggested to be done, and maybe ask some questions?
If all goes well, I hope for well-received pull requests from my side which may need some support from yours to become contributions :wink:
No ETA though (I am from Rhineland, so time is always on my side, if I only knew which one it is).

**Comment by hkolbeck** [comment]
Both of your help is very welcome! If you have any questions, I'm more than happy to answer them.

A good starting point is probably to go over changes to the spec since the last update and add test cases as described [Here](https://github.com/hkolbeck/kdl4j/tree/trunk/src/test/resources).
